src/ctia/http/middleware/unknown.clj

```diff
@@ -0,0 +1,18 @@
+(ns ctia.http.middleware.unknown
+  (:require [ring.util.response :refer [not-found]]))
+
+(defn- err-html
+  "Construct an HTML body for a 404 response."
+  [uri]
+  (str "<html> <head> <meta http-equiv=\"Content-Type\" content=\"text/html;charset=ISO-8859-1\"/>"
+       " <title>Error 404 </title> </head> "
+      "<body> <h2>HTTP ERROR: 404</h2> <p>Problem accessing "
+       uri
+       ". Reason: <pre>    Not Found</pre></p> </body> </html>"))
+
+(defn wrap-unknowns
+  "When a request results in a nil response, then this indicates that no route handled it."
+  [handler]
+  (fn [request]
+    (or (handler request)
+        (not-found (err-html (:uri request))))))
```

Comment:
Compojure API provides a ["not-found" handler](https://github.com/metosin/compojure-api/wiki/Routing#not-found-at-runtime). I think it's a better option that respect response format. I don't know if it applies to the dependency that requires this change.

Reply:
I removed the middleware, and added a `compojure.route/not-found` path as suggested.

I have kept the unknown.clj file (along with the now-unused `wrap-unknowns` wrapper), as this contains the HTML to be returned. A reason to keep the `wrap-unknowns` function is if we ever want to have HTML that looks like the previous version returned by ring/compojure. The old 404 response included the requested URI, but the `compojure.route/not-found` path does not get access to this.